fiftyone/core/stages.py

```diff
@@ -941,8 +942,8 @@ def _get_meta_filtered_fields(sample_collection, meta_filter, frames=False):
             base, leaf = key.split(".", 1)
             info_filter[leaf] = val
 
-    matcher = (
-        lambda q, v: q.lower() in v.lower()
+    matcher = lambda q, v: (
```

Comment:
I find this lambda incredibly confusing. I recommend refactoring to a named function instead
```python

  def matcher(query, value):
    if isinstance(query, str):
      if isinstance(value, str):
        return query.lower() in value.lower()
      elif isinstance(value, dict):
        return query.lower() in str(value).lower()
    return query == value
```